File: CCAuction.Domain/Models/ItemBidHistory.cs
using CCAuction.Domain.Interfaces;
using System;

namespace CCAuction.Domain.Models
{
    public class ItemBidHistory : IId, IDateCreated
    {
        public Guid Id { get; set; }
        public Guid ItemId { get; set; }
        public Item Item { get; set; }
        public decimal BidPrice { get; set; }
        public DateTimeOffset BidDate { get; set; }
        public Guid BidderId { get; set; }
        public User Bidder { get; set; }
        public DateTimeOffset DateCreated { get; set; }
    }
}

File: CCAuction/ClientApp/src/app/models/itemBidHistory.ts
import { User } from '../models/user';
export class ItemBidHistory {
  id: string;
  itemId: string;
  bidPrice: number;
  bidDate: Date;
  bidderId: string;
  bidder: User;

  constructor (newItemId : string, newBidPrice : number, newBidDate: Date, newBidderId: string, newBidder: User) {
    this.itemId = newItemId;
    this.bidPrice = newBidPrice;
    this.bidDate = newBidDate;
    this.bidderId = newBidderId;
    this.bidder = newBidder;
  }
}
